src/dbnode/client/replicated_session.go

```diff
@@ -81,15 +75,11 @@ func withNewSessionFn(fn newSessionFn) replicatedSessionOption {
 }
 
 func newReplicatedSession(opts Options, options ...replicatedSessionOption) (clientSession, error) {
-	// TODO(srobb): Replace with PooledWorkerPool once it has a GoIfAvailable method
```

Comment:
There was a conversation about this on the original PR: https://github.com/m3db/m3/pull/1859/files#diff-c1dc6d5d77e2b9ee46fe6cb2deff3e36R85

The `Go` method on `PooledWorkerPool` blocks if a worker isn't available. Better to keep using a `WorkerPool` until the `GoIfAvailable` method is added?

Reply:
I see. thanks for the context. will change back to WorkerPool